### PR TITLE
Revert "`memoryview`: remove inheritance from `Sequence`"

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Reverts python/typeshed#12781

Although it does not implement the full Sequence API at runtime, memoryview is in fact a subclass of Sequence at runtime (through ABC registration). I will work soon on changes in CPython to remedy the discrepancy, but that doesn't solve the problem on existing versions of Python.

I'd prefer to revert to the previous behavior where memoryview is a subclass of Sequence, for a few reasons:

- It's been like this for a long time and hasn't led to a lot of issues. Both options introduce potential problems; it's better to stick with the existing solution in case people are relying on it in their code already.
- Removing the inheritance makes it so the behavior of `isinstance(..., Sequence)` becomes inconsistent between type checkers and the runtime. That's confusing and can lead to soundness issues.
- I don't find the arguments in https://github.com/python/typeshed/pull/12781#issuecomment-2407802965 convincing. The missing methods are relatively rarely used, note core parts of the Sequence API. Sequence may not appear in the MRO of `memoryview`, but that's hos we handle ABC inheritance throughout typeshed. There wasn't fallout in mypy-primer, but most typechecked code isn't in mypy-primer. Inheritance from ABCs can lead to metaclass problems in type checkers, but removing the ABC from a single builtin class hardly fixes that problem.

### CODE REVIEW

**Comment by Avasam** [comment]
Is it possible to override the missing methods in a way that makes them unusable / almost always raise typing errors? Maybe something like a NoReturn + deprecated + Never args?

**Comment by JelleZijlstra** [comment]
Done. I set index and count to None as @AlexWaygood suggested in the other thread. I didn't attempt to set `__contains__` and `__reversed__` to None because while those specific methods don't exist on memoryview, the operations they represent do work.

**Review by AlexWaygood — APPROVED** [review]
I like it much more with the `None` overrides, and your point about this causing unsound type narrowing with `isinstance()` is a good one. I still think you're underrating the impact of IDEs and autocomplete tools suggesting nonexistent "public-API" methods in autocompletions (and failing to error when the user accesses them in their code). But fixing it in CPython is clearly the correct long-term fix (thank you for taking up that cause!).

TL;DR: still not sure it's holistically better overall to have it inherit from Sequence, but fine with it being reverted, especially with the `None` overrides for `index`/`count`
